### ISSUE TITLE
Source Snapchat Marketing: adaccounts_stats_daily stream sync fails with `expected string or bytes-like object`

### ISSUE THREAD

**Issue by monik-a** [airbytehq/airbyte#20180]
## Environment
- **Airbyte version**:  example is 0.40.22
- **Deployment**: Kubernetes using official charts
- **Source Connector and version**: Snapchat 0.1.8 
- **Destination Connector and version**: Snowflake 0.4.40
- **Step where error happened**: Sync job 
## Current Behavior
I reset the stream and resync the whole thing - it runs perfectly :white_check_mark:
It’s scheduled every day at midnight and it runs with a success the next day, syncing all streams. :white_check_mark:
The following day it fails with: "Additional Failure Information: expected string or bytes-like object" . The problem is coming from the source.
This means that every 2 days I’d have to reset and resync… :octavia-shock:
I’m attaching the exemplary log and the setup of the connector in the coment below :point_down:

## Expected Behavior
Stream should run every day the same way - it runs only once after resync, which doesn't make much sense.

## Logs

I'll attach logs in a comment. Otherwise they're in the Slack thread: https://airbytehq.slack.com/archives/C021JANJ6TY/p1670239051928729?thread_ts=1670238832.791479&cid=C021JANJ6TY

## Steps to Reproduce
As in current behaviour:
- Set up the sync scheduled with cron
- run the initial sync
- wait for the first scheduled run to succeed
- wait for the second scheduled run to fail.

**Comment by monik-a** [comment]
[dc7dd622_3d4a_4164_bbf2_eb4e35990748_logs_https://github.com/airbytehq/airbyte/files/10174921/dc7dd622_3d4a_4164_bbf2_eb4e35990748_logs_244_txt.1.txt)

![image (4)](https://user-images.githubusercontent.com/40922847/206150372-ceb9d6d0-ac83-4695-8caf-19e572456232.png)
![image (3)](https://user-images.githubusercontent.com/40922847/206150378-00a179bb-dacd-4ea2-bd95-864a0194862c.png)
244_txt (1).txt](

**Comment by sh4sh** [comment]
Hey @monik-a thanks for reporting this. To confirm that this isn't caused by the most recent changes, would you be willing to try an older version of the connector Source Snapchat Marketing 0.1.6? I'm not sure if this will help, but knowing those changes didn't cause this behaviour will help us figure out where to look.

**Comment by monik-a** [comment]
I changed to 0.1.6 and it still failed, just with the persist error:
`Failure Origin: persistence, Message: Something went wrong during state persistence`
[dc7dd622_3d4a_4164_bbf2_eb4e35990748_logs_287_txt.txt](https://github.com/airbytehq/airbyte/files/10183921/dc7dd622_3d4a_4164_bbf2_eb4e35990748_logs_287_txt.txt)

**Comment by sh4sh** [comment]
@monik-a I think resetting the sync may fix that issue if you're willing to try!

**Comment by monik-a** [comment]
As I explained, the dance I've been doing for the past 2 weeks is:
- reset the sync - the full resync works
- the next scheduled sync works too
- the next one fails, so I'm back to step 1.

Is there something else I need to do for the reset to work and stay?
Apologies if I misunderstood something.

**Comment by sh4sh** [comment]
Hey @monik-a, thanks so much for reporting this issue and for your help investigating this. We have reproduced the bug and discovered that in _some_ cases the source is emitting an incorrect state for `*_daily` streams in incremental mode. This issue has been escalated to the Airbyte oncall team and prioritized. As the investigation moves forward I'll continue sharing updates here, feel free to reach out or tag me here if you notice any related behaviour or have any questions.

**Comment by sh4sh** [comment]
@monik-a this was fixed in the Snapchat Marketing source in v0.1.9, can you try the latest version to see if it helps? https://github.com/airbytehq/airbyte/pull/20498

**Comment by sh4sh** [comment]
Closing this issue, please report a new issue if anything related arises with source-snapchat-marketing:v0.1.9 or later
